src/features/сart/CartSlice.ts

```diff
@@ -0,0 +1,90 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Product } from '@/types/Phone';
+
+interface CartItem extends Product {
+  quantity: number;
+  clickedBuy: boolean;
+  clickedFav?: boolean;
+}
+
+interface CartState {
+  items: Record<string, CartItem>;
+}
+
+const initialState: CartState = {
+  items: {},
+};
+
+const cartSlice = createSlice({
+  name: 'cart',
+  initialState,
+  reducers: {
+    addToCart: (state, action: PayloadAction<Product>) => {
+      const product = action.payload;
+      const existingItem = state.items[product.id];
+
+      const updatedItem = existingItem
+        ? {
+            ...existingItem,
+            quantity: existingItem.quantity + 1,
+            clickedBuy: true,
+          }
+        : { ...product, quantity: 1, clickedBuy: true };
+
+      return {
+        ...state,
+        items: {
+          ...state.items,
+          [product.id]: updatedItem,
+        },
+      };
+    },
+
+    updateQuantity: (
+      state,
+      action: PayloadAction<{ itemId: string; quantity: number }>,
+    ) => {
+      const { itemId, quantity } = action.payload;
+
+      if (!state.items[itemId]) {
+        return state;
+      }
+
+      return {
+        ...state,
+        items: {
+          ...state.items,
+          [itemId]: { ...state.items[itemId], quantity },
+        },
+      };
+    },
```

Comment:
If you added an 'Enter' in line 42, I think it would be good to put it here as well. Just for readability.